m4/ax_check_openssl.m4

```diff
@@ -1,5 +1,5 @@
 # ===========================================================================
```

Comment:
Other than copying over the ax_check_openssl from the autoconf-achive, what changed or wasn't working with my copy of the file?

Reply:
The only interesting fix is https://github.com/autoconf-archive/autoconf-archive/commit/cf5911f8ce107845940d4145bb8ab8f0ce49c2da to improve cross support (using the right macro to find pkg-config). The other 2 changes are cosmetic (HTTPSify a link and tweak a message: https://github.com/autoconf-archive/autoconf-archive/commit/571886d8488be24f8473b461070b86b944534ef6).